plugins/org.jkiss.dbeaver.ext.oracle/src/org/jkiss/dbeaver/ext/oracle/model/OracleSchema.java

```diff
@@ -929,7 +929,7 @@ protected JDBCStatement prepareObjectsStatement(JDBCSession session, OracleSchem
                          + " c\r\n"
                          + "LEFT JOIN " 
                          + OracleUtils.getAdminAllViewPrefix(session.getProgressMonitor(), getDataSource(),"CONSTRAINTS") + " rc\r\n"
-                         + "ON rc.OWNER = c.r_OWNER AND rc.CONSTRAINT_NAME = c.R_CONSTRAINT_NAME\r\n"
+                         + "ON rc.OWNER = ? AND rc.CONSTRAINT_NAME = c.R_CONSTRAINT_NAME\r\n"
```

Comment:
Be careful, this change is a change in semantics. You are effectively saying `rc.OWNER = c.OWNER` instead of `rc.OWNER = c.r_OWNER`. That is because you write `rc.OWNER = ?` and `c.OWNER = ?` and for both you set `OracleSchema.this.getName()` as a bind parameter.
If this is a referential constraint, the results will not be the same. So unless the previous code had a bug, you are introducing one.

Reply:
It seems OK for me. Code before patch:
```
                         + "LEFT JOIN " 
                         + OracleUtils.getAdminAllViewPrefix(session.getProgressMonitor(), getDataSource(),"CONSTRAINTS") + " rc\r\n"
                         + "ON **rc.OWNER = c.r_OWNER** AND rc.CONSTRAINT_NAME = c.R_CONSTRAINT_NAME\r\n"
                         + "WHERE\r\n" + "    c.CONSTRAINT_TYPE = 'R'\r\n" + "    AND **c.OWNER = ?**\r\n"
                         + "    AND c.TABLE_NAME = ?");
```

after patch:
```
                         + "LEFT JOIN " 
                         + OracleUtils.getAdminAllViewPrefix(session.getProgressMonitor(), getDataSource(),"CONSTRAINTS") + " rc\r\n"
                         + "ON **rc.OWNER = ?** AND rc.CONSTRAINT_NAME = c.R_CONSTRAINT_NAME\r\n"
                         + "WHERE\r\n" + "    c.CONSTRAINT_TYPE = 'R'\r\n" + "    AND c.OWNER = ?\r\n"

```
=>  `c.owner=?` in the WHERE statement can be propagated to left join and `*rc.OWNER = c.r_OWNER` can be finally replaced by `rc.OWNER = ?`

Reply:
I see it now...

Reply:
There could be cross schema constraint in the Oracle, but then the second commit cannot be used and the query is 6-10x slower, but definitely faster than original one. Thanks for feedback.